packages/xstate-wallet/src/store/channel-store-entry.ts

```diff
@@ -127,7 +136,28 @@ export class ChannelStoreEntry {
         firstState.appData === secondState.appData
       );
     }
-    return true;
+
+    // JS implementation of App check
+    const fromVariablePart = getVariablePart(toNitroSignedState(firstState)[0].state);
+    const toVariablePart = getVariablePart(toNitroSignedState(secondState)[0].state);
+    const turnNumB = secondState.turnNum;
+
+    const ConsensusAppArtifact = require('@statechannels/nitro-protocol/build/contracts/ConsensusApp.json');
```

Comment:
Is there a way to make this channel-agnostic? Is it crazy for the app to provide the artifact to the wallet?

Reply:
Previously I had envisioned that app definitions would be like chrome extensions. A user must specifically accept / acknowledge adding a new one to their wallet, but the wallet might also come pre-packaged with some trusted ones

Reply:
So, the user can accept rules that are:
- provided by some URL, or the app, or whatever
- deployed at address `0x123`

The wallet fetches the contract from the URL or whatever, and verifies the bytecode of the contract matches the bytecode deployed at `0x123` in the selected blockchain?

Reply:
Yeah